nexus/src/db/model/network_interface.rs

```diff
@@ -85,17 +85,17 @@ pub struct NetworkInterfaceUpdate {
     pub description: Option<String>,
     pub time_modified: DateTime<Utc>,
     #[diesel(column_name = is_primary)]
-    pub make_primary: Option<bool>,
+    pub primary: Option<bool>,
 }
 
 impl From<params::NetworkInterfaceUpdate> for NetworkInterfaceUpdate {
     fn from(params: params::NetworkInterfaceUpdate) -> Self {
-        let make_primary = if params.make_primary { Some(true) } else { None };
+        let primary = if params.primary { Some(true) } else { None };
```

Comment:
I guess this probably predates this change, but I think having optional booleans with default value "false" kind of sucks because a lot of client code treats falsey (like `null` or `undefined` or otherwise missing) as false.  I'd be all for making these required.

Reply:
Can you explain a bit more why that's a problem? A value of false is effectively ignored in the implementation, since (1) if the interface is a secondary, then that'd be a no-op, and (2) setting the primary to a secondary isn't supported, one has to set a secondary to the new primary. It seems like the client sending a value of false or sending nothing both have the intended behavior in that case.

Reply:
I see, thanks for the explanation. I was focused on this particular API, but yes, I agree in general it might be better to just avoid this type of interface.